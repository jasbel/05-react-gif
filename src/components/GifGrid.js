import React, {  useEffect, useState } from 'react'
import GifGridItem from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';
import useFetchGifs from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);

    const {data: images, loading } = useFetchGifs(category);
    console.log(loading);

    // useEffect(() => {
    //     getGifs( category )
    //         .then( setImages );
    // }, [ category ])

    

    return (
        <>
            <h3 className="animate__animated animate__fadeIn" > { category } </h3>

            { loading && <p className="animate__animated animate__flash">loading...</p>}

            <div className="card-grid">
                {
                    images.map(gif =>
                        <GifGridItem 
                            key={gif.id}
                            {...gif}
                        />
                        
                    )
                }
            </div>
        </>
    )
}

export default GifGrid
