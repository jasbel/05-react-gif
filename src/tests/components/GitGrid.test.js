import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("pruebas en >GitGrid", () => {
    const category = "One Punch";

    test("should de mostrarse correctament", () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test("debe de mostrar items cuando se cargan las imagines de useFetchGifs", () => {
        const gifs = [
            { id: "ABC", url: "https://localsf/pfoot.jpg", title: "mi title" },
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length );
    });
});
