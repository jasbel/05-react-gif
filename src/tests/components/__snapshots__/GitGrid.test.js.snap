// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en >GitGrid debe de mostrar items cuando se cargan las imagines de useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="mi title"
      url="https://localsf/pfoot.jpg"
    />
  </div>
</Fragment>
`;

exports[`pruebas en >GitGrid should de mostrarse correctament 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
